src/api/cmlibs/zinc/element.h

```diff
@@ -187,6 +187,32 @@ ZINC_API int cmzn_element_get_dimension(cmzn_element_id element);
 ZINC_API cmzn_elementfieldtemplate_id cmzn_element_get_elementfieldtemplate(
 	cmzn_element_id element, cmzn_field_id field, int componentNumber);
 
+/**
+ * Get the number of faces for this element. This is how many faces its shape
+ * should have, irrespective of whether faces are defined or whether they are
+ * collapsed by a coordinate field. Note: 1-D line elements do not have faces.
+ *
+ * @param element  The element to query.
+ * @return  Number of faces for element's shape, or 0 if none or failed due to
+ * invalid element or if element shape is undefined.
+ */
+ZINC_API int cmzn_element_get_number_of_faces(cmzn_element_id element);
```

Comment:
Is it possible to return a negative number to indicate an error?  How is a person to distinguish between 0 faces and an error?

Reply:
Good idea. Implementing now.